libplatsupport/CMakeLists.txt

```diff
@@ -34,10 +34,20 @@ config_choice(
     "ega;LibPlatSupportX86ConsoleDeviceEGA;LIB_PLAT_SUPPORT_SERIAL_TEXT_EGA;KernelPlatPC99"
 )
 
+if(NOT ${KernelArmMach} STREQUAL "")
```

Comment:
Maybe we should change the logic to check platform settings first and fall back on the (to be deprecated) `KernelArmMach` only if nothing is set here. 

```
set(LibPlatSupportMach "")
if(KernelPlatformRpi3)
    set(LibPlatSupportMach "bcm283x")
elseif(KernelPlatformRpi4)
    set(LibPlatSupportMach "bcm283x")
else if(NOT ${KernelArmMach} STREQUAL "")
    # falling back to kernel settings is done to keep legacy compatibility.
    set(LibPlatSupportMach "${KernelArmMach}")
endif()
```

Reply:
Just implemented proposed version, seems to compile.